src/components/PhotoFrame.tsx

```diff
@@ -112,7 +112,7 @@ const PhotoFrame = ({
     const updateRequired = useRef(false);
 
     const [filteredData, setFilteredData] = useState<EnteFile[]>([]);
-
+    const hasPersonalFiles = files.length - incomingShareFiles.length > 0;
```

Comment:
I meant that we could have the state `hasPersonalFiles`, instead of doing the calculation on each render.